nucliadb_core/src/query_planner.rs

```diff
@@ -146,25 +174,71 @@ impl From<SearchRequest> for QueryPlan {
 }
 
 fn compute_pre_filters(search_request: &SearchRequest) -> Option<PreFilterRequest> {
-    let Some(timestamp_filters) = search_request.timestamps.as_ref() else {
-        return None;
-    };
     let mut pre_filter_request = PreFilterRequest {
         timestamp_filters: vec![],
+        labels_filters: vec![],
     };
+
+    // Timestamp filters
+    let request_has_timestamp_filters = search_request.timestamps.as_ref().is_some();
+    if request_has_timestamp_filters {
+        let timestamp_filters = compute_timestamp_pre_filters(search_request);
+        pre_filter_request
+            .timestamp_filters
+            .extend(timestamp_filters);
+    }
+
+    // Labels filters
+    let request_has_labels_filters = search_request.filter.is_some()
```

Comment:
This is clone is not needed since you only want to check if `field_labels` is empty. Also, maybe the expression is a bit too complex and has hidden logic (it needs lazy evaluation). How about this:
```rust
    let request_has_labels_filters = search_request
        .filter
        .as_ref()
        .map(|i| i.field_labels.is_empty())
        .unwrap_or_default();
```